models/branches.go

```diff
@@ -38,6 +38,11 @@ func (protectBranch *ProtectedBranch) BeforeUpdate() {
 	protectBranch.UpdatedUnix = time.Now().Unix()
 }
 
+// IsProtected returns if the branch is protected
+func (protectBranch *ProtectedBranch) IsProtected() bool {
+	return protectBranch.ID > 0
```

Comment:
@daviian you can change here to `protectBranch != nil && protectBranch.ID > 0` and than you can remove that nil check above

Reply:
@lafriks Current implementation would reflect current master. But if you want I can of course add the check inside the function.

Reply:
@daviian ok, let it be this way than for now, this could be optimized in master than